site/_posts/2020-09-03-oidc-auth-api-usage.md

```diff
@@ -0,0 +1,173 @@
+
+The steps are:
+1. Request the JWT Token from the OpenID-Connect provider
+1. Retrieve the access_token from the JWT
+1. (OPTIONAL) Use the access_token to do Introspection
+1. Accessing MiQ API using the access_token in the header
+
+The bash example commands:
+
+```bash
+# Accessing the ManageIQ API with the access_token from a JWT Token
+
+   # Step 1: Request the JWT Token from the OpenID-Connect provider
+   jwt_token=$(curl -k -L --user ${user}:${password} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "grant_type=password" -d "client_id=${oidc_client_id}" -d "client_secret=${oidc_client_secret}" -d "username=${user}" -d "password=${password}" ${oidc_provider_token_endpoint} )
+
+   # Step 2: Retrieve the access_token from the JWT
+   access_token=$(echo $jwt_token | jq -r '.access_token')
+
+   # Step 3 (OPTIONAL) Use the access_token to do Introspection
+   curl -k -L --user ${oidc_client_id}:${oidc_client_secret} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "token=${access_token}" ${oidc_auth_introspection_endpoint}
+
+
+   # Step 4: Accessing MiQ API using the access_token in the header
+   curl -L -vvv -k -X GET -H "Authorization: Bearer ${access_token}" https://${miq_server_name}/api/users | jq
+```
+
+
+## Example: Using a ManageIQ API Auth Token
+---------------------------------------------------------------------
+
+This example details the steps for using a ManageIQ API Auth Token for a non-admin user.
+
+The steps are:
+1. Request the JWT Token from the OpenID-Connect provider
+1. Retrieve the access_token from the JWT
+1. (OPTIONAL) Use the access_token to do Introspection
+1. Request an MiQ API Authentication Token using the access_token in the header:
+1. Retrieve the API authentication token from the result
+1. Accessing MiQ API Using the MiQ API Auth Token
+
+The bash example commands:
+
+```bash
+# Accessing the ManageIQ API using a ManageIQ API Auth Token
+
+   # Step 1: Request the JWT Token from the OpenID-Connect provider
+   jwt_token=$(curl -k -L --user ${user}:${password} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "grant_type=password" -d "client_id=${oidc_client_id}" -d "client_secret=${oidc_client_secret}" -d "username=${user}" -d "password=${password}" ${oidc_provider_token_endpoint} )
+
+   # Step 2: Retrieve the access_token from the JWT
+   access_token=$(echo $jwt_token | jq -r '.access_token')
+
+   # Step 3: (OPTIONAL) Use the access_token to do Introspection
+   curl -k -L --user ${oidc_client_id}:${oidc_client_secret} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "token=${access_token}" ${oidc_auth_introspection_endpoint}
+
+   # Step 4: Request an MiQ API Authentication Token:
+   result=$(curl -L -vvv -k -X GET -H "Authorization: Bearer ${access_token}" https://${miq_server_name}/api/auth)
+
+   # Step 5: Retrieve the API authentication token from the result
+   api_auth_token=`echo $result | jq -r '.auth_token'`
```

Comment:
combine 4 and 5,  api_auth_token=$(curl ... | jq -r '.auth_token')

Reply:
I'd like to leave this as it. Although note efficient code, I think it makes for clearer documentation.
If you don't feel strongly I'd like to leave this as two steps.